src/features/mutation_minigame/nodes/dna/DNA.gd

```diff
@@ -0,0 +1,54 @@
+extends "res://features/dynamicmenu/DynamicMenu.gd"
+var base_scene = load("res://features/mutation_minigame/nodes/dna/Nucleotide.tscn")
+
+
+func add_base(base_str="BLANK") -> void:
+	var base = base_scene.instantiate()
+	base.base = base_str
+	base.mutation.connect(mutation_handler)
+
+
+func random_dna(length: int):
+	var rng = RandomNumberGenerator.new()
+	var bases: Array = ["A","G","T","C"]
+	
+	var blank_start = base_scene.instantiate()
+	blank_start.mutation.connect(mutation_handler)
+	self.add_child(blank_start)
+	for i in range(length):
+		var base = base_scene.instantiate()
+		base.base = bases[rng.randi_range(0, bases.size()-1)]
+		base.mutation.connect(mutation_handler)
+		self.add_child(base)
+		var blank = base_scene.instantiate()
+		blank.mutation.connect(mutation_handler)
+		self.add_child(blank)
+		
+		
+		
+	
+func mutation_handler(node: Control, data: Dictionary):
+	if data["Type"] == "Deletion":
+		self.hide_slide_child(node)
+		self.hide_slide_child(self.get_child(node.get_index() + 1))
```

Comment:
I'm unsure if there is a better way, but we always have this `base` & `blank` setup. Is there a way to join these together so we don't need to use indices programmatically? I.e., we can just `hide_slide_child` on a single node.

Reply:
The problem is that sometimes the mutation acts on a `base` (deletion), and other times it acts on a `blank` (insertion). Not entirely sure the best way to rectify this either, but I'll give it some thought.

 I just realized I have to implement substitution as well. I *was* pretty tired when I wrote this.